pkg/registry/node/etcd/etcd.go

```diff
@@ -102,3 +103,25 @@ var _ = rest.Redirector(&REST{})
 func (r *REST) ResourceLocation(ctx api.Context, id string) (*url.URL, http.RoundTripper, error) {
 	return node.ResourceLocation(r, r.connection, r.proxyTransport, ctx, id)
 }
+
+// An interface for types that allow getting information about a Node on which give pod is running.
+type HostLocator interface {
```

Comment:
The problem is that we actually need to read data from etcd to get the Port for a given Kubelet, as it's no longer 'static' (in a sense that each Kubelet may be able to choose it's own port to run on), while we have a single 'KubeletClient' in master. We'll need to remove Port from it eventually, but it needs to stay there for backward compatibility.